src/com/dfbnc/sockets/UserSocket.java

```diff
@@ -943,111 +921,50 @@ private void processNonAuthenticated(final String[] line) {
                 sendIRCLine(Consts.ERR_NOTREGISTERED, line[0], "You must login first.");
         }
 
-        // After every message, check if we have everything we need...
-        if (realname != null && password != null && nickname != null) {
-            authenticate(line[0]);
-        }
+        checkAuthStatus(line[0]);
     }
 
-    /**
-     * Attempts to authenticate the user based on the information they've provided.
-     *
-     * @param lastCommand The command the user last attempted, used in error responses.
-     */
-    private void authenticate(final String lastCommand) {
-        final String[] bits = username.split("\\+");
-        username = bits[0];
-        clientID = (bits.length > 1 && !bits[1].isEmpty()) ? bits[1].replaceAll("[^a-z0-9_-]", "") : null;
-        if (bits.length > 2 && !bits[2].isEmpty()) {
-            clientType = ClientType.getFromName(bits[2].toLowerCase());
```

Comment:
(In theory the bouncer is supposed to auto-detect this, and it tries based on the CTCP version - but after I added that bit, I learned that tapchat sucks and doesn't version reply.)